controllers/client/openstackclient_controller.go

```diff
@@ -360,14 +373,108 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 }
 
+// fields to index to reconcile when change
+const (
+	caSecretNameField          = ".spec.caSecretName"
+	openStackConfigMapField    = ".spec.openStackConfigMap"
+	openStackConfigSecretField = ".spec.openStackConfigSecret"
+)
+
+var (
+	allWatchFields = []string{
+		caSecretNameField,
+		openStackConfigMapField,
+		openStackConfigSecretField,
+	}
+)
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *OpenStackClientReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
+	// index caSecretName
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &clientv1.OpenStackClient{}, caSecretNameField, func(rawObj client.Object) []string {
```

Comment:
OK so the field indexer based solution looks better than what we have today in nova-operator where we list all the secret and filter in golang https://github.com/openstack-k8s-operators/nova-operator/blob/0578e9c7d47dd718b46eacd1be4bfbf49bb54285/controllers/common.go#L442